diff --git a/gcn/utils.py b/gcn/utils.py
index 67f4b96..2580381 100644
--- a/gcn/utils.py
+++ b/gcn/utils.py
@@ -65,16 +65,16 @@ def load_data(dataset_str):
         ty_extended[test_idx_range-min(test_idx_range), :] = ty
         ty = ty_extended
 
-    features = sp.vstack((allx, tx)).tolil()
+    features = sp.vstack((x, tx)).tolil() # CHANGED
     features[test_idx_reorder, :] = features[test_idx_range, :]
     adj = nx.adjacency_matrix(nx.from_dict_of_lists(graph))
 
-    labels = np.vstack((ally, ty))
+    labels = np.vstack((y, ty)) # CHANGED
     labels[test_idx_reorder, :] = labels[test_idx_range, :]
 
     idx_test = test_idx_range.tolist()
-    idx_train = range(len(y))
-    idx_val = range(len(y), len(y)+500)
+    idx_train = range(len(y)) # CHANGED
+    idx_val = range(test_idx_range.tolist()[-1], labels.shape[0]) # CHANGED
 
     train_mask = sample_mask(idx_train, labels.shape[0])
     val_mask = sample_mask(idx_val, labels.shape[0])
